logging/testing/util.go

```diff
@@ -27,13 +27,19 @@ import (
 )
 
 // TestLogger gets a logger to use in unit and end to end tests
-func TestLogger(t *testing.T) *zap.SugaredLogger {
+func TestLogger(i interface{}) *zap.SugaredLogger {
 	opts := zaptest.WrapOptions(
 		zap.AddCaller(),
 		zap.Development(),
 	)
 
-	return zaptest.NewLogger(t, opts).Sugar()
+	switch v := i.(type) {
+	case *testing.B:
+		return zaptest.NewLogger(v, opts).Sugar()
+	case *testing.T:
+		return zaptest.NewLogger(v, opts).Sugar()
```

Comment:
Should we panic on default? That should never happen, right?

Reply:
Currently  all e2e, UT, benchmark etc... pass `testing.B` or `testing.T` only
But still have written empty object at the end if both case fails

Reply:
I'd let it panic 🤔. Nothing else is supposed to reach here and we have a bug in the tests if it does.